common/OpTestInstallUtil.py

```diff
@@ -0,0 +1,312 @@
+import threading
+import SocketServer
+import BaseHTTPServer
+import SimpleHTTPServer
+import commands
+import time
+from Exceptions import CommandFailed
+import OpTestConfiguration
+
+
+BASE_PATH = ""
+INITRD = ""
+VMLINUX = ""
+KS = ""
+DISK = ""
+USERNAME = ""
+PASSWORD = ""
+REPO = ""
+BOOTPATH = ""
+conf = OpTestConfiguration.conf
+
+
+class InstallUtil():
+    def __init__(self, base_path="", initrd="", vmlinux="",
+                 ks="", boot_path="", repo=""):
+        global BASE_PATH
+        global INITRD
+        global VMLINUX
+        global KS
+        global DISK
+        global USERNAME
+        global PASSWORD
+        global BOOTPATH
+        global REPO
+        self.conf = conf
+        self.host = conf.host()
+        self.system = conf.system()
+        self.system.host_console_unique_prompt()
+        self.console = self.system.sys_get_ipmi_console()
+        self.server = ""
+        self.repo = conf.args.os_repo
+        REPO = self.repo
+        DISK = self.host.get_scratch_disk()
+        USERNAME = self.host.username()
+        PASSWORD = self.host.password()
+        BOOTPATH = boot_path
+        BASE_PATH = base_path
+        INITRD = initrd
+        VMLINUX = vmlinux
+        KS = ks
+
+    def wait_for_network(self):
+        retry = 30
+        while retry > 0:
+            try:
+                output = self.console.run_command("ifconfig -a")
+                return True
+            except CommandFailed as cf:
+                if cf.exitcode is 1:
+                    time.sleep(1)
+                    retry = retry - 1
+                    pass
+                else:
+                    raise cf
+
+    def assign_ip_petitboot(self):
+        """
+        Assign host ip in petitboot
+        """
+        self.console.run_command("stty cols 300")
+        self.console.run_command("stty rows 30")
+        # Lets reduce timeout in petitboot
+        self.console.run_command("nvram --update-config petitboot,timeout=10")
+        cmd = "ip addr|grep -B1 -i %s|grep BROADCAST|awk -F':' '{print $2}'" % self.conf.args.host_mac
+        iface = self.console.run_command(cmd)[0].strip()
+        cmd = "ifconfig %s %s netmask %s" % (iface, self.host.ip, self.conf.args.host_submask)
+        self.console.run_command(cmd)
+        cmd = "route add default gateway %s" % self.conf.args.host_gateway
+        self.console.run_command_ignore_fail(cmd)
+        cmd = "echo 'nameserver %s' > /etc/resolv.conf" % self.conf.args.host_dns
+        self.console.run_command(cmd)
+
+    def get_server_ip(self):
+        """
+        Get IP of server where test runs
+        """
+        my_ip = ""
+        self.wait_for_network()
+        # Check if ip is assigned in petitboot
+        try:
+            self.console.run_command("ping %s -c 1" % self.conf.args.host_gateway)
+        except CommandFailed as cf:
+            self.assign_ip_petitboot()
+            self.console.run_command("ping %s -c 1" % self.conf.args.host_gateway)
```

Comment:
Can we have the check for n/w ping be done in a loop for some time? As i have seen some instances we are running ping command before actually the n/w configuration takes into account(ifconfig and route commands taking some time to have the effect).
```
 ERROR [177.065s]: runTest (testcases.InstallUbuntu.InstallUbuntu)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins_data/sath/pridhivi/satheesh_tests/op-test-framework/testcases/InstallUbuntu.py", line 95, in runTest
    my_ip = OpIU.get_server_ip()
  File "/home/jenkins_data/sath/pridhivi/satheesh_tests/op-test-framework/common/OpTestInstallUtil.py", line 118, in get_server_ip
    self.console.run_command("ping %s -c 1" % self.conf.args.host_gateway)
  File "/home/jenkins_data/sath/pridhivi/satheesh_tests/op-test-framework/common/OpTestIPMI.py", line 328, in run_command
    raise CommandFailed(command, res, exitcode)
CommandFailed: Command 'ping x.xx.xxx.x -c 1' exited with 1.
```
After that i re-checked the n/w is actually UP, so op-test little running faster before the n/w configures in the host.

Reply:
@pridhiviraj done..